pyanaconda/kickstart_dispatcher.py

```diff
@@ -0,0 +1,359 @@
+    """
+
+    # file name to be used in case of parsing string if not supplied
+    unknown_filename = "<MAIN>"
+
+    def __init__(self, handler, valid_sections=None, missing_include_is_fatal=True):
+        """Initialize the parser.
+
+        :param valid_sections: list of valid section names (including '%')
+        :type valid_sections: list(str)
+        :param missing_include_is_fatal: raise KickstartError if included file
+                                         is not found
+        :type missing_include_is_fatal: bool
+        """
+
+        self._valid_sections = valid_sections or []
+        # calls setupSections
+        super().__init__(handler, missingIncludeIsFatal=missing_include_is_fatal)
+        self._current_ks_filename = self.unknown_filename
+        self._result = TrackedKickstartElements()
+
+    @property
+    def valid_sections(self):
+        """List of valid kickstart sections"""
+        return list(self._valid_sections)
+
+    @valid_sections.setter
+    def valid_sections(self, value):
+        self._valid_sections = value
+
+    def split(self, filename):
+        """Split the kickstart file into elements.
+
+        :param filename: name of kickstart file
+        :type filename: str
+
+        :return: object containing kickstart elements with references to
+                 kickstart files
+        :rtype: KickstartElements
+        """
+        with open(filename, "r") as f:
+            kickstart = f.read()
+        return self.split_from_string(kickstart, filename=filename)
+
+    def split_from_string(self, kickstart, filename=None):
+        """Split the kickstart given as string into elements.
+
+        :param kickstart: kickstart to be split
+        :type kickstart: str
+        :param filename: filename to be used as file reference in the result
+        :type filename: str
+
+        :return: object containing kickstart elements with references to
+                 kickstart
+        :rtype: KickstartElements
+        """
+        self._reset()
+        self._current_ks_filename = filename or self.unknown_filename
+        self.readKickstartFromString(kickstart)
+        return self._result
+
+    def add_section(self, section):
+        """Adds a StoreSection to the result."""
+        if section.args[0] == "%addon":
+            try:
+                name = section.args[1]
+            except IndexError:
+                name = ""
+            self._result.append_addon(name=name,
+                                      content=str(section),
+                                      header_lineno=section.header_lineno,
+                                      filename=self._current_ks_filename)
+        else:
+            self._result.append_section(name=section.args[0],
+                                        content=str(section),
+                                        header_lineno=section.header_lineno,
+                                        filename=self._current_ks_filename)
```

Comment:
You can set `name` to `section.args[0]` in the `else` branch and join the calls of `self._result.append_section`.

Reply:
I really hate these args solution it's so hard to find what means `section.args[0]` and `section.args[1]`.

Reply:
Yes, I should consolidate this.